### ISSUE TITLE
File Versioning for Non-Encrypted Files

### ISSUE THREAD

**Issue by AryasDad** [nextcloud/end_to_end_encryption#247]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.

### Steps to reproduce
1. Turn on E2EE
2. Encrypt ONLY one folder
3.

### Expected behaviour
Tell us what should happen
File Versioning should not be affected by folders not using E2EE

### Actual behaviour
Tell us what happens instead
File versioning doesn't work for ANY file or folder

### Server configuration

**Operating system:**
Ubuntu Ubuntu 20.04.2 LTS

**Web server:**
Apache/2.4.41 (Ubuntu)

**Database:**
mysql  Ver 15.1 Distrib 10.3.29-MariaDB

**PHP version:**
PHP 7.4.3 (cli) 

**Nextcloud version:** (see Nextcloud admin page)
22.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh install

**Where did you install Nextcloud from:**
nextcloud.com

**Signing status:**
<details>
<summary>Signing status</summary>
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

Enabled:
  - accessibility: 1.7.0
  - activity: 2.15.0
  - admin_audit: 1.11.0
  - bruteforcesettings: 2.2.0
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - dav: 1.18.0
  - encryption: 2.9.0
  - end_to_end_encryption: 1.7.1
  - federatedfilesharing: 1.11.0
  - files: 1.16.0
  - files_pdfviewer: 2.3.0
  - files_rightclick: 1.1.0
  - files_sharing: 1.13.2
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.11.0
  - firstrunwizard: 2.11.0
  - groupfolders: 9.0.2
  - logreader: 2.7.0
  - lookup_server_connector: 1.9.0
  - nextcloud_announcements: 1.11.0
  - notifications: 2.10.1
  - oauth2: 1.9.0
  - password_policy: 1.12.0
  - privacy: 1.6.0
  - provisioning_api: 1.11.0
  - serverinfo: 1.12.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - support: 1.5.0
  - survey_client: 1.10.0
  - text: 3.3.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.1
  - twofactor_gateway: 0.19.0
  - updatenotification: 1.11.0
  - viewer: 1.6.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0
Disabled:
  - circles
  - contactsinteraction
  - dashboard
  - federation
  - files_external
  - photos
  - recommendations
  - systemtags
  - user_ldap
  - user_status
</details>

**Are you using encryption:** yes/no
Yes

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

**Comment by tobiasKaminsky** [comment]
What exactly do you mean with "file versioning does not work"?
Are there any info/error in nextcloud.log?

**Comment by AryasDad** [comment]
Nothing in nextcloud.log file indicating any info/error.

"file versioning does not work" means that when I edit a file and then go and look for versioning online or in the iOS app, nothing appears.  

I have since moved onto another solution of using Cryptomator to encrypt my folder and turned E2EE completely off.  I just did a test on a text file in a non-encrypted folder and now can see it is getting "versioned" online.  However, do not see versioning within iOS app, so need to figure out if I am just no looking in correct place for it.

Just to recap, E2EE appears to disable versioning for the entire next cloud installation, irregardless of it is being actively used to encrypt a single folder.  All other non-encrypted files do not get "versioning".  I understand the stated limitations for file versioning not working with E2EE, as the server cannot decrypt the E2EE folder.  

HOWEVER, if a folder is NOT using E2EE, it should from a pure opinionated logical standpoint, still get versioning.  If versioning is indeed disabled for the ENTIRE Nextcloud account, then I think the documentation should be updated to clarify this that it removes this function for the entire installation and not just the folder/file it is being used on.

Would be nice to not have to use two separate apps to handle encryption on one folder, but if this is how E2EE is suppose to work, then I will have to stick with my alternative solution.

**Comment by karlemilnikka** [comment]
I can confirm that this issue persists with Nextcloud 24.0.4.

**Comment by tobiasKaminsky** [comment]
@CarlSchwan do you have an idea why this could happen?

**Comment by karlemilnikka** [comment]
I can confirm that this issue persists with Nextcloud 25.0.2.

**Comment by joshtrichards** [comment]
Are you by chance also utilizing Server-side Encryption at the same time as E2EE? That might trigger this behavior (and also isn't a supported configuration).

**Comment by karlemilnikka** [comment]
@joshtrichards Yes. I installed the server with server-side encryption and when i tested the E2EE app, the server-side encryption wasn’t disabled.

**Comment by karlemilnikka** [comment]
@joshtrichards After disabling server-side encryption, versioning seems to work alongside the E2EE app.
